dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -1182,46 +1104,64 @@ def dpnp_qr_batch(a, mode="reduced"):
         [a_copy_ev],
     )
 
-    ht_lapack_ev.wait()
-    a_ht_copy_ev.wait()
+    ht_list_ev = [ht_geqrf_batch_ev, a_ht_copy_ev]
 
-    if mode == "r":
-        r = a_t[..., :k].swapaxes(-2, -1)
-        out_r = dpnp.triu(r).astype(res_type, copy=False)
-        return out_r.reshape(batch_shape + out_r.shape[-2:])
-
-    if mode == "raw":
-        q = a_t.reshape(batch_shape + a_t.shape[-2:])
-        r = tau_h.reshape(batch_shape + tau_h.shape[-1:])
-        return (
-            q.astype(res_type, copy=False),
-            r.astype(res_type, copy=False),
-        )
+    if mode in ["r", "raw"]:
+        dpctl.SyclEvent.wait_for(ht_list_ev)
```

Comment:
why do we need a wait here? we are passing depended event into `_triu_inplace` below, then it makes no sense.